pymc_extras/statespace/models/structural/components/regression.py

```diff
@@ -194,64 +206,110 @@ def make_symbolic_graph(self) -> None:
             row_idx, col_idx = np.diag_indices(self.k_states)
             self.ssm["state_cov", row_idx, col_idx] = sigma_beta.ravel() ** 2
 
-    def populate_component_properties(self) -> None:
+    def _set_parameters(self) -> None:
         k_endog = self.k_endog
         k_endog_effective = 1 if self.share_states else k_endog
+        k_states = self.k_states // k_endog_effective
+
+        beta_param_name = f"beta_{self.name}"
+        beta_param_shape = (k_endog_effective, k_states) if k_endog_effective > 1 else (k_states,)
+        beta_param_dims = (
+            (f"endog_{self.name}", f"state_{self.name}")
+            if k_endog_effective > 1
+            else (f"state_{self.name}",)
+        )
```

Comment:
Let's eliminate the intermediate variables and just put these directly into the `Parameter` constructor